src/common/hook/useQueue/utils/publish.tsx

```diff
@@ -97,7 +97,33 @@ export const getRawDocuments = async (forms: FormEntry[], config: Config): Promi
 };
 
 const wrapDocuments = async (rawDocuments: any[]) => {
-  return utils.isRawV3Document(rawDocuments[0]) ? await wrapDocumentsV3(rawDocuments) : wrapDocumentV2(rawDocuments);
+  return utils.isRawV3Document(rawDocuments[0]) ? await wrapDocumentsV3(rawDocuments) : wrapDocumentsV2(rawDocuments);
+};
+
+const processVerifiableDocument = async (
+  jobs: PublishingJob[],
+  nonce: number,
+  contractAddress: string,
+  verifiableDocuments: RawDocument[]
+) => {
+  if (verifiableDocuments.length === 0) return;
+  console.log(verifiableDocuments);
+  const rawDocuments = verifiableDocuments.map((doc) => doc.rawDocument);
+  const wrappedDocuments = await wrapDocuments(rawDocuments);
+  const firstWrappedDocument = wrappedDocuments[0];
+  const merkleRoot = utils.getMerkleRoot(firstWrappedDocument);
+  const firstRawDocument = verifiableDocuments[0];
+  jobs.push({
+    type: firstRawDocument.type,
+    nonce,
+    contractAddress,
+    documents: verifiableDocuments.map((doc, index) => ({
+      ...doc,
+      wrappedDocument: wrappedDocuments[index],
+    })),
+    merkleRoot: merkleRoot,
+    payload: {},
+  });
```

Comment:
dont mutate in a separate function, returns the jobs array and build the collection again later

Reply:
done